service/matching/task.go

```diff
@@ -42,10 +42,10 @@ type (
 		decisionTaskInfo *types.MatchingPollForDecisionTaskResponse
 		activityTaskInfo *types.PollForActivityTaskResponse
 	}
-	// internalTask represents an activity, decision, query or started (received from another host).
+	// InternalTask represents an activity, decision, query or started (received from another host).
 	// this struct is more like a union and only one of [ query, event, forwarded ] is
 	// non-nil for any given task
-	internalTask struct {
+	InternalTask struct {
```

Comment:
Trying to understand why we need to export this type. Looks like an internal type definition for matching package to me?

Reply:
Yes. Other components in matching are already exported so I make this struct exported as well. My plan is to do a refactor in matching (similar to the history, to group the components in sub packages and export them)